app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -298,20 +305,44 @@ private void updateNavigationMenuSelection() {
         }
     }
 
-    private void addRoomColumns(HorizontalSnapScrollView scroller) {
-        LinearLayout root = (LinearLayout) scroller.getChildAt(0);
-        root.removeAllViews();
-        if (scroller.getColumnWidth() != 0) {
-            // update pre-calculated width with actual layout
-            columnWidth = scroller.getColumnWidth();
+    private void addRoomColumns(@NonNull HorizontalSnapScrollView horizontalScroller, @NonNull List<Lecture> lectures, int roomCount, boolean forceReload) {
+        int roomIndexLeft = horizontalScroller.getColumn();
+        int roomIndexRight = roomIndexLeft + maxRoomColumnsVisible;
+
+        // whenever possible, just update recycler views
+        if (!forceReload && !adapterByRoom.isEmpty()) {
+            for (int roomIndex = roomIndexLeft; roomIndex <= roomIndexRight; roomIndex++) {
+                //noinspection ConstantConditions
+                adapterByRoom.get(roomIndex).notifyDataSetChanged();
+            }
+            return;
         }
-        for (int i = 0; i < MyApp.roomCount; i++) {
-            LinearLayout roomLayout = new LinearLayout(context);
-            LinearLayout.LayoutParams p = new LayoutParams(
-                    columnWidth, LayoutParams.MATCH_PARENT, 1);
-            roomLayout.setOrientation(LinearLayout.VERTICAL);
-            roomLayout.setLayoutParams(p);
-            root.addView(roomLayout);
+
+        LinearLayout columnsLayout = (LinearLayout) horizontalScroller.getChildAt(0);
+        columnsLayout.removeAllViews();
+        adapterByRoom.clear();
+
+        int boxHeight = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
+        LayoutCalculator layoutCalculator = new LayoutCalculator(Logging.Companion.get(), boxHeight);
+        LecturesByRoom lecturesByRoom = new LecturesByRoom(lectures);
+
+        for (int roomIndex = 0; roomIndex <= roomCount; roomIndex++) {
+            List<Lecture> roomLectures = lecturesByRoom.get(roomIndex);
+
+            LinearLayoutManager columnLayout = new LinearLayoutManager(context);
```

Comment:
:mag: Not sure if this works but you could try to move the `LinearLayoutManager` outside the loop and instantiating it once, only.

Reply:
Me neither, I trust SO at that point https://stackoverflow.com/questions/36766682/one-linearlayoutmanager-is-only-for-one-recyclerview ;)

Reply:
Sound reasonable. How about inlining the variable at its usage?